### ISSUE TITLE
[Template] - Class request

### ISSUE THREAD

**Issue by Wngui** [Wngui/CS2WarcraftMod#1]
_Please use this template for requesting new classes_

**Class name:** [Class]

**Normal abilities**
1. [Ability name] - [description]
2. [Ability name] - [description]
3. [Ability name] - [description]

**Ultimate ability**
[Ultimate name] - [description]

**Comment by baskerdk** [comment]
Faceless Void
Deagle and knife only. Max 2 per team,

Time Walk
15/20/25/30/35/40% chance to evade.

Backtrack (ability)
save your current HP, after 1/1/2/2/3/4 seconds set your HP back to that value. this fails if you die. 8 second cooldown

Time Lock
4/8/12/16/20/24% chance on hit to prevent the enemy from shooting or moving for 1 second.

Chronosphere (ultimate)
Lay a 150/200/250/300/300/300 unit purple ward that lasts 6/8/9/10/11/12 seconds. any player, teammate or enemy, who isn't playing Faceless Void that enters the ward area cannot move, shoot, attack, aim, or activate abilities or ultimates. Faceless Void players are forced to use their knife while they are in the ward area. all players are immune to Deagle damage while they are in the ward area. 40 second cooldown. this pierces ward immunity but not ultimate immunity.

**Comment by baskerdk** [comment]
Faerie Dragon

Reinforced Hide
you spawn with 5/10/15/20/25/30/35/40 more HP

Mana Flare
4/8/12/16/20/24/28/32% chance when hit to blind the enemy for 4 seconds with a 100% opacity light yellow screen.

Phase Shift
4/8/12/16/20/24/28/32% to evade, when you evade become 100% invisible for 1 second.

Dream Rift (ultimate)
become 100% invisible and fly for 9/10/11/12/13/14/15/16 seconds. your flying and invisibility ends when you are shot, even if you evade. 25 second cooldown.

**Comment by baskerdk** [comment]
SpongeBob 

4 skills / 5 levels each

Yummy Yummy: 20-60% chance to refill one bullet in your current clip when shot.

Shock Absorber: 20-100% chance to reduce incoming headshot damage by 10-60%.

Slippery Sponge (Ability): Get 200-300% movement speed boost for 1-3s.

Ultimate Sponge (Ultimate): Become a sponge for 1-2s during which all incoming damage is converted to health. Max HP: 200.

Information:
Player turns green during Ultimate Sponge

**Comment by baskerdk** [comment]
Undead Scourge
4 skills 

Vampiric Aura: 10-30% chance to leech 20-75% hp on attack.

Unholy Aura: Get x1.1-x1.6 speed.

Levitation: Get x0.85-x0.50 gravity.

Suicide Bomber: On death, explode to deal 30-90 dmg to players within 100-300 units.

Restricted Items:
Boots of Speed
Winged Boots

**Comment by baskerdk** [comment]
Vampire Prince Level 550+

Knife race

5 skills / 4 levels each

Blood suck: 10/25/40/55% chance on hit to leech 50% of the damage you dealt.

Undead Agility: Spawn with 120/130/140/150% increased movement speed.

Shadow Meld: Spawn with 30/40/50/60% invisibility.

Battle of Honor: 10/20/30/40% chance on hit to force the enemy to drop their weapon.

Bat Form: (Ultimate) Throw yourself forward with a force of 2000/2250/2500/2750

**Comment by baskerdk** [comment]
Diablos

has a black model. requires total level 32

Gravija
12/16/20/24/28/32% chance on hit to deal 75% extra damage.

Poison
2/4/6/8/10/12% chance on hit to poison the enemy for 5 seconds. they take 5 damage per second for that duration.

Darkside
8/16/24/32/40/48% chance on hit to blind the enemy with a 50% opacity black screen for 5 seconds.

Osmose
You spawn with 15/20/25/30/35/40 more HP and 4/8/12/16/20/24% increased movement speed.

Dark Messenger
50/60/70/80/90/100% chance when you die to create a black hole at the location you died at 2 seconds after you die. the black hole is an 800 unit radius thick black outlined ward with a white circle at the center of it that can be seen through walls. the black outlines are slowly rotating. The ward constantly pulls all enemies on the map towards it with a force that only allows the fastest races to not lose ground to it or a teleportation skill to escape it. if the black hole is off the ground enemies can be pulled off the ground up into it. the black hole makes a whooshing noise the entire time it is active. if any player, teammate or enemy steps into the ward area they are instantly killed. the black hole lasts 6 seconds. when this procs, everyone gets a warning message in chat. "A Dark Messenger approaches...".

ultimate immunity stops the pull of the black hole, but does not prevent you from dying if you enter its radius.

**Comment by baskerdk** [comment]
**Raid Leader**
Limited to SMGs, knife, nades.
**Skills**
**1. Focus Fire (Ability 1)**
Mark a random enemy player to receive 3/5/7/9 more damage from all bullets.

*- 5 second cooldown to use once round starts, 10 second cooldown on use.
- Once marked, all chat states: "[Player] has been marked for focus fire!"
- The debuff on the marked player lasts until either: (1) the round ends, (2) the currently marked player dies, or (3) the raid leader uses Focus Fire ability and the mark is applied to a different player.*

**2. Healing Check (Ability 2)**
Restore 50 armor to all allied players. If any allied player has 100 armor, restore 3/5/7/9 health to that player instead.

*- 15 second cooldown to use once round starts, 15 second cooldown on use.
- Armor restoration caps out at 100.
- Healing Check also applies to the Raider Leader who uses it.*

**3. Over There! (Ability 3)**
Reveal a random enemy player's location by spawning a VFX on them. Cooldown is 60/50/40/30 seconds. 

*- Over There! spawns a circular visual effect that quickly expands throughout the map from where the enemy is standing. 
- If the enemy has lowered transparency, 20% chance to set their visibility to 100% visible for rest of the round.*

**4. Min-Maxed (Passive)**
Receive helmet + 25/50/75/100 armor upon spawn, move at 105/110/115/120% speed, and have 105/110/115/120 health.

**5. Boss Mechanics Coach (Ult)**
Mark a random allied player to survive the next bullet that would kill them. Their health is set to 1. Cooldown is 60/50/40/30 seconds.

*- 10 second cooldown to use once round starts.
- The buff on the marked player lasts until either: (1) the round ends, or (2) the currently marked player dies.
- The buff cannot proc on a player that is already marked with the buff, the skill should try to mark someone else.
- In other words, the mark can be (re)applied to multiple teammates throughout a round but cannot stack on itself.
- Boss Mechanics Coach can apply to the Raid Leader who uses it.*

**Comment by baskerdk** [comment]
Admiral Darnaw

Rage
Your armor is your rage. you have 30 maximum rage, if you ever have more than your maximum rage, set your rage to its maximum instead. You lose 10 rage per second if you have not been hit or gained rage in the last .75/1/1.25/1.5/1.75/2 seconds. you gain 3 rage when you hit an enemy. you deal 1% more damage with hits for each point of rage you have. you do not lose armor when you are hit.

Admiral's Arrogance
10/12/14/16/18/20% chance that when you would gain rage from a hit that you gain your maximum rage instead.

Vengeful Cry (ability)
Cannot be used unless you have taken damage in the last 3/4/5/6/7/8 seconds. For the next 4.5 seconds, all teammates (not you) in a 450 unit radius take 50% reduced damage. for that duration you cannot lose rage, you gain 3 rage per enemy in that radius up to a maximum of 25 rage gained, and your max rage is increased to 55. 28/26/24/22/20/18 second cooldown.

Warlord's Mark (ability2)
Enemies in a 450 unit radius are marked. While marked, all hits against them grant 3% life steal. Marked enemies who are bashed have a 5/10/20/30/40/50% chance to be bashed for double the duration. Enemies who are bashed by you grant you 10 rage. 20/18/16/14/12/10 second cooldown.

Exploit Weakness
Enemies with damage reduction instead take that much increased damage from your hits. Example, if an enemy has 35% damage reduction they instead take 35% MORE damage from you. 4/8/12/16/20/25% chance on hit when you hit an enemy that is not burning, stunned, or shocked, to burn, stun, or shock them. If you burn them, they are burned for 3.5 seconds. if you stun them, bash them for 1 second. if you shock them, they take 10% additional damage from hits for 6 seconds.

Berserk (Ultimate) 1 level
while active you lose 5 rage per second. each second this is active you lose 1 more rage per second. while this is active you also deal 1% more damage with hits for each point of rage you have. (this stacks with the first skill) you lose 5 rage when you activate this ultimate and must have at least 5 to activate. you cannot turn this ultimate off, it can only turn off when you run out of rage. no cooldown. you cannot have negative rage, if you would set your rage to 0.

**Comment by baskerdk** [comment]
Death Knight

requires total level 40

Death Coil (ability)
Deal 5/10/15/20/25 damage to the enemy you are aiming at. 10 second cooldown.

Frostmourne
10/20/30/40/50% chance on hit to slow the enemy by 20% for 4 seconds. this stacks with itself.

Unholy Aura
spawn with 6/12/18/24/30/36% increased movement speed.

Animate Dead (ultimate)
respawn a random teammate. 45/40/35/30/25/20 second cooldown.

**Comment by baskerdk** [comment]
Lux
5Skills/8 Levels each:

**__Illumination__**
Extra damage and chance to illuminate (basically a damage mark that is visible on the player model - was a blue aura back then)
Hits have a Chance to Apply Marks, Hits then have a chance to consume mark for 25 extra damage at max

**__Light Binding__**
Chance to freeze victim (dont remember numbers, chance wasn't that right and freeze was maybe 1 second)

**__Prismatic Barrier:__**
Heal yourself and teammates over time
1hp/level every 10 seconds
AoE Over Time heal to self/ nearby 8hp when maxed

**__Lucent Singularity:__**
Use ability1 to place a ward at your crosshair
Slows(duration 1 second?) and applies illumination 

Levels probably increased the slow%

**__Finales Funkeln (Ultimate)__**
Use ultimate to fire a rocket, applies 1 Illumination and immediately consumes it

**Comment by baskerdk** [comment]
**__Night Elves__**
*4 skills / 8 levels each*

**__Evasion__**: 10-30% chance to evade a shot.

**__Thorns Aura__**: 5-40% chance to deal 30% mirror damage.

**__Trueshot Aura__**: 30% chance to deal 110-160% dmg.

**__Entangling Roots__** *(Ultimate)*: Roots all enemies within 100-600 units for 2-6s. (30s cooldown)

**Comment by baskerdk** [comment]
Priest

3 skills / 8 levels each

Holy Life. Get 20-150 extra HP.

Holy Shield. 5-20% chance to deal 10-30% mirror damage.

Curing Ritual. (Ultimate) Get pushed where you aim with a force of 1000-3000.

**Comment by baskerdk** [comment]
Wolverine

Skill 1: Temper
- Grants health + Speed

Skill 2: Adamantium Bones
- Immune to knife damage (100% at max level)

Skill 3: Adamantium Claws
- Chance to deal increased damage

Skill 4: Heal Factor
- When being shot at you have a chance to heal yourself

Skill 5: Violent Outburst (ultimate)
- Gain 100% chance to evade all damage for 5 seconds

**Comment by pZyk93** [comment]
Going to suggest the classic War3FT classes.

### Undead Scourge:

- Vampiric Aura
"You recieve 10/20/30/40% of your damage dealt as health"

- Unholy Aura
"You run 10/20/30/40% faster"

- Levitation Aura
"You can jump higher - Less gravity"

- Ultimate: Suicide Bomber
"You explode when you die, can be manually activated"

_______________

### Human Alliance:

- Invisibility
"Makes you partially invisible, increasing each level"

- Devotion Aura
"Gives you an additional 15/25/35/45 health."

- Bash
"7/13/19/25% chance to bash the enemy.
Renders enemy immobile for a short duration"

- Ultimate: Teleport
"Teleport toward where you aim."

_____________

### Orcish Horde:

- Critical Strike
"Chance of doing critical damage with weapons."

- Critical Grenade
"Grenades will do 40-240% more damage"

- Reincarnation
"Gives you 15-80% chance of respawning once"

- Ultimate: Chain Lightning
"Discharges a bolt of lightning that jumps to nearby enemies, dealing damage"

_____________

### Night Elf:

- Evasion
"5/10/15/20% chance of evading a shot"

- Thorns Aura
"You deal 5/10/15/20% of damage received to your attacker"

- Trueshot Aura
"Your attacks deal 5/10/15/20% more damage"

- Ultimate: Entangling Roots
"Bind enemies to the ground, rendering them immobile for 1.25/1.5/1.75/2 seconds"

____________

### Blood Mage:

- Phoenix
"20-50% chance to revive your teammates that die.
Each time you revive, chance is reduced by half to a minimum of 2-8%"

- Banish
"20% chance of making an enemy blind and disoriented for a short duration"

- Siphon Mana
"8% chance to steal 1% of an enemys money on damage"

- Ultimate: Flame Strike
"Burn the enemy over time for 5 damage 5 times."

_____________

### Shadow Hunter:

- Healing Wave
"Heal teammates around you"

- Hex
"Chance of resisting other enemy's skill attacks"

- Serpent Wards
"Wards that damage enemies if they touch them. Activates with +ability"
 
- Ultimate: Big Bad Voodoo
"You are invulnerable from physical attacks for a duration"

_____________

### Warden:

- Fan of Knives
"Chance of becoming a mole, infiltrating the enemy team"

- Immunity
"Grants immunity from enemy ultimates"

- Shadow Strike
"Chance of poisoning your enemy"

- Ultimate: Vengeance
"Respawn upon first death of each round"

____________

### Crypt Lord:

- Impale
"Attacks grants a chance of causing the opponents crosshair to shake"

- Spiked Carapace
"Reduce incoming damage, reflecting some of the damage back to your attacker"

- Carrion Beetles
"Chance of dealing extra damage"

- Ultimate: Locust Swarm
"Release a locust swarm that damages opponents"
